server/src/main/java/org/apache/iotdb/db/utils/datastructure/TVList.java

```diff
@@ -275,11 +249,18 @@ protected void clearSortedTime() {
   protected void checkExpansion() {
     if ((size % ARRAY_SIZE) == 0) {
       expandValues();
-      timestamps.add(
-          (long[]) PrimitiveArrayPool.getInstance().getPrimitiveDataListByType(TSDataType.INT64));
+      timestamps.add((long[]) getDataListByType(TSDataType.INT64));
     }
   }
 
+  protected Object getDataListByType(TSDataType dataType) {
+    return PrimitiveArrayManager.getDataListByType(dataType);
+  }
+
+  public boolean checkIfDataListIsEnough(int lengthToBeAdded) {
+    return size + lengthToBeAdded < ARRAY_SIZE;
```

Comment:
Is this right?  If the ARRAY_SIZE is 128, and current TVList size is 129 (contains two array), then even add one data point, this returns false

Reply:
Yes, make sense. This is a bug exactly.....